src/bin/compress.rs

```diff
@@ -76,43 +79,58 @@ fn main() {
     // assert!(!out_dir_p.exists());
     // std::fs::create_dir(out_dir_p).unwrap();
     
-    let (mut programs, tasks, num_prior_inventions) = args.fmt.load_programs_and_tasks(&args.file).unwrap();
+    let (mut train_programs,
+        test_programs,
+        tasks,
+        num_prior_inventions) = args.fmt.load(&args.file, args.split_train_test).unwrap();
     
     if args.shuffle {
-        programs.shuffle(&mut rand::thread_rng());
+        train_programs.shuffle(&mut rand::thread_rng());
     }
     if let Some(n) = args.truncate {
-        programs.truncate(n);
+        train_programs.truncate(n);
     }
     
     // parse the program strings into expressions
-    let programs: Vec<Expr> = programs.iter().map(|p| p.parse().unwrap()).collect();
+    let train_programs: Vec<Expr> = train_programs.iter().map(|p| p.parse().unwrap()).collect();
+    let test_programs: Option<Vec<Expr>> = if let Some(ps) = test_programs { Some(ps.iter().map(|p| p.parse().unwrap()).collect()) } else { None };
+    
 
     // for prog in programs.iter() {
     //     println!("{}", prog);
     // }
     println!("{}","**********".blue().bold());
     println!("{}","* Stitch *".blue().bold());
     println!("{}","**********".blue().bold());
-    programs_info(&programs);
+    println!("{}","* Training programs: *".blue().bold());
+    programs_info(&train_programs);
+    println!("{}","* Testing programs: *".blue().bold());
+    if let Some(test_ps) = &test_programs {
+        programs_info(test_ps);
+    } else {
+        println!{"No test programs given"}
+    }
 
     // build a single `Expr::Programs` node from these programs. Stitch uses these because often we want to treat
     // different parts of the same programs the same way that we treat different parts of different programs, so
     // treating everything as one big expression makes sense.
-    let programs: Expr = Expr::programs(programs);
+    let train_programs: Expr = Expr::programs(train_programs);
+    let test_programs: Option<Expr> = if let Some(ps) = test_programs { Some(Expr::programs(ps)) } else { None };
 
-    if programs.to_string_curried(None).contains("(app (lam") {
+    if train_programs.to_string_curried(None).contains("(app (lam")  {
```

Comment:
This should check in `test_programs` too but given that you can't combine `if-let`s and `if`s on stable it would be super verbose and ugly so I left it for now...